lib/checkPullRequestTemplate.js

```diff
@@ -0,0 +1,187 @@
+      return line.startsWith('2');
+    });
+
+    const explanationText = '[Explain here what your PR does and why]';
+    if (explanation.includes(explanationText)) {
+      // The user has not added the explanation.
+      return (
+        'The body of this PR is missing the required description, ' +
+        'please update the body with a description of what this PR does.'
+      );
+    }
+    return '';
+  } else {
+    // There is no overview section in the body of the PR.
+    return (
+      'The body of this PR is missing the overview section, please ' +
+      'update it to include the overview.'
+    );
+  }
+};
+
+/**
+ * checks that the pull request's body contains the explanation.
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const checkEssentialChecklist = (pullRequest) => {
+  const checklistText = '## Essential Checklist';
+  const checklistHeadingIndex = pullRequest.body.indexOf(checklistText);
+
+  if (checklistHeadingIndex !== -1) {
+    const nextHeadingIndex = pullRequest.body.indexOf(
+      '## ',
+      checklistHeadingIndex + checklistText.length
+    );
+
+    let checklistSection = pullRequest.body.substring(
+      checklistHeadingIndex,
+      nextHeadingIndex
+    );
+    const uncheckedItems = checklistSection.split('\n').filter((list) => {
+      return list.startsWith('- [ ]');
+    });
+
+    let message = '';
+
+    uncheckedItems.forEach((item) => {
+      if (item.toLowerCase().includes('linter/karma')) {
+        // Karma and linter is unchecked.
+
+        message +=
+          'The karma and linter checklist has not been checked, ' +
+          'please make sure to run the frontend tests and lint tests before ' +
+          'pushing. ';
+      }
+      if (item.toLowerCase().includes('allow edits from maintainers')) {
+        if (!pullRequest.maintainer_can_modify) {
+          message +=
+            'The allow edits from maintainers checklist needs to ' +
+            'be ticked so that maintainers can rerun failed tests. Endeavour ' +
+            'to add this by ticking on the check box. ';
+        }
+      }
+    });
+
+    return message;
+  } else {
+    return (
+      'The body of this PR is missing the checklist section, please ' +
+      'update it to include the checklist.'
+    );
+  }
+};
+
+/**
+ * @param {import('probot').Context} context
+ */
+const checkTemplate = async (context) => {
+  /**
+   * @type {import('probot').Octokit.PullsGetResponse} pullRequest
+   */
+  const pullRequest = context.payload.pull_request;
+
+  const explanationMessage = checkPRExplanation(pullRequest);
+  const checklistMessage = checkEssentialChecklist(pullRequest);
+  let commentBody = '';
+  if (explanationMessage && checklistMessage) {
+    commentBody =
+      'Hi @' +
+      pullRequest.user.login +
+      ', ' +
+      swapCharInPlace(
+        explanationMessage,
+        0,
+        explanationMessage[0].toLowerCase()
```

Comment:
Why not return with the first char as lowercase instead?

Reply:
Done.

Reply:
Do we need the swap first char function now?